### ISSUE TITLE
[FEATURE] Install Harvester on RAID1 partition

### ISSUE THREAD

**Issue by iosifnicolae2** [harvester/harvester#3909]
**Is your feature request related to a problem? Please describe.**
Having the operating system installed on a RAID 1 partition is a must for us.

**Describe the solution you'd like**
It would be great to be able to select two disks when installing Harvester, and then setup a RAID 1 configuration.

**Describe alternatives you've considered**
I've tried to install Harvester normally and then convert the main partition to a RAID 1 partition.. it's pretty complicated to do so..

**Comment by iosifnicolae2** [comment]
If you have any ideas on how can we do this feel free to leave a comment.

Thank you!

**Comment by iosifnicolae2** [comment]
Normally, it shouldn’t be very complicated to implement it but unfortunetly I don't have the context & expertise for this.

Here is what I've found:
  - here we will need to remove grep -w disk - https://github.com/harvester/harvester-installer/blob/76ee43a83eb97d1d7bab5803b2067626860b3a4d/pkg/console/install_panels.go#L338
- then, we will need to not tear down LVM and MD (maybe display a warning or something like this) - https://github.com/harvester/harvester-installer/blob/76ee43a83eb97d1d7bab5803b2067626860b3a4d/package/harvester-os/files/usr/sbin/harv-install#L440
- from what I’ve tested elemental install delete /dev/md0 partition, not sure why.. - https://github.com/harvester/harvester-installer/blob/76ee43a83eb97d1d7bab5803b2067626860b3a4d/package/harvester-os/files/usr/sbin/harv-install#LL445C1-L445C18

I hope that someone from Rancher could pick this up or give us some suggestions on how to implement this.

**Comment by xeruf** [comment]
I would very much like this to be implemented as well, am now setting up my servers which have two redundant system disks and cannot properly utilize them :/

**Comment by xeruf** [comment]
Also, what about installing on an intel RST-managed RAID1?

**Comment by slyviacassell** [comment]
Install harvester on intel RST-managed RAID1 array is quite nice! It would make the best use of some old server!

**Comment by air-awan** [comment]
Agree, I've been using Intel RST with other hypervisor/Linux distribution.

**Comment by FelipeMiranda** [comment]
For sure this is very welcome feature, in my case I have 2 ssd's that I would like to use in RAID1, I'm moving from proxmox to harvester and in proxmox we have the option to setup zfs in raid mode. This is essential as we can't rely on the system to not have a fail safe for the system disk.
